version: "3.8"
services:
    database:
      image: mariadb:latest
      environment:
        MYSQL_ROOT_PASSWORD: CHANGE_ME
        MYSQL_DATABASE: CHANGE_ME
        MYSQL_USER: CHANGE_ME
        MYSQL_PASSWORD: CHANGE_ME
      restart: on-failure
      volumes:
        - ${PWD}/mariadb:/var/lib/mysql
      healthcheck:
        test: ["CMD", "healthcheck.sh", "--connect", "--innodb_initialized"]
        start_period: 10s
        interval: 10s
        timeout: 5s
        retries: 3        
      networks:
        - database_network
        
    website:
        build: .
        volumes:
            - ./hksova:/usr/src/app/hksova
            - ./configs/default.py:/usr/src/app/configs/default.py:ro
            - ./configs/docker.py:/usr/src/app/configs/docker.py:ro
            - ./configs/uwsgi/docker_wsgi.ini:/usr/src/app/configs/uwsgi/docker_uwsgi.ini:ro
        networks:
            - nginx_network
            - database_network
        restart: always
        depends_on:
            database:
                condition: service_healthy

    adminer:
      image: adminer
      restart: always
      ports:
        - 8080:8080
      networks:
        - database_network

    nginx:
        image: nginx:mainline-alpine
        volumes:
            - ./configs/nginx/docker_hksova.conf:/etc/nginx/conf.d/default.conf:ro
            - /etc/letsencrypt:/etc/letsencrypt/:ro
        ports:
            - 8081:80
            - 10443:443
        depends_on:
            - website
        networks:
            - nginx_network

networks:
    nginx_network:
        driver: bridge
    database_network:
        driver: bridge

